crate/hsm/base_hsm/src/session/session_impl.rs

```diff
@@ -1214,6 +1677,17 @@ impl Session {
         {
             return Ok(None);
         }
+        let key_type = match self.get_key_type(object_handle)? {
+            None => return Ok(Some(id)),
+            Some(key_type) => key_type,
+        };
+        let term = "_pk".as_bytes();
+        if id.ends_with(term) {
+            return Ok(Some(id))
+        }
+        if key_type == KeyType::RsaPublicKey {
+            id.append(&mut term.to_vec())
+        }
```

Comment:
Eventually try this:
```suggestion
        if let Some(KeyType::RsaPublicKey) = self.get_key_type(object_handle)? {
            if !id.ends_with(b"_pk") {
            id.extend_from_slice(b"_pk");
            }
        }
```